test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")) {
```

Comment:
This does not help much cause it does not verify only mutating command can make change. Hence, you can add 'else' here to guarantee non-mutating commands do not save (e.g. Checking the updated time of file)